ledger/block/src/transaction/deployment/mod.rs

```diff
@@ -105,6 +105,16 @@ impl<N: Network> Deployment<N> {
         Ok(u64::try_from(self.to_bytes_le()?.len())?)
     }
 
+    /// Returns the number of program functions in the deployment.
+    pub fn len(&self) -> usize {
```

Comment:
is `len` making it obvious that this refers to the number of functions in the program? otherwise `num_functions` might be a better pick

Reply:
Indeed its a bit of an odd naming without context, though it is not unnatural. The `Deployment` contains just one vector which equals the number of functions/verifying_keys/certificates.

I mainly did this to have consistent naming with `Execution::len`, where it refers to the number of transitions. This PR then uses both `deployment.len()` and `execution.len()` to determine at which "index" to insert an additional Fee transition.

Reply:
Overall, I'd opt for a more precise naming for clarity. But this is fine as well.